test/README.md

```diff
@@ -6,53 +6,91 @@ This directory contains tests and testing docs for `Knative Eventing`.
   they test
 - [End-to-end tests](#running-end-to-end-tests) reside in [`/test/e2e`](./e2e)
 
-## Running unit tests
+## Running tests with scripts (not recommended)
 
-Use `go test`:
+### Presubmit tests
+
+[`presubmit-tests.sh`](./presubmit-tests.sh) is the entry point for the tests
+before code submission.
+
+You can run it simply with:
 
 ```shell
-go test -v ./pkg/...
+test/presubmit-tests.sh
 ```
 
-_By default `go test` will not run [the e2e tests](#running-end-to-end-tests),
-which need [`-tags=e2e`](#running-end-to-end-tests) to be enabled._
+_By default, this script will run `build tests`, `unit tests` and
+`integration tests`._ If you only want to run one type of tests, you can run this
+script with corresponding flags like below:
+
+```shell
+test/presubmit-tests.sh --build-tests
+test/presubmit-tests.sh --unit-tests
+test/presubmit-tests.sh --integration-tests
+```
+
+_Note that if the tests you are running include `integration tests`, it will
+create a new GKE cluster in project `$PROJECT_ID`, start Knative Serving and
+Eventing system, upload test images to `$KO_DOCKER_REPO`, and run all
+`e2e-*tests.sh` scripts under [`test`](.). After the tests finish, it will
+delete the cluster._
 
-## Presubmit tests
+### E2E tests
 
-[`presubmit-tests.sh`](./presubmit-tests.sh) is the entry point for the
-[end-to-end tests](/test/e2e).
+[`e2e-tests.sh`](./e2e-tests.sh) is the entry point for running all e2e tests.
 
-This script, and consequently, the e2e tests will be run before every code
-submission. You can run these tests manually with:
+You can run it simply with:
 
 ```shell
-test/presubmit-tests.sh
+test/e2e-tests.sh
+```
+
+_By default, it will create a new GKE cluster in project `$PROJECT_ID`, install
+Knative Serving stack with Istio, upload test images to your Docker repo and run
+the end-to-end tests against the Knative Eventing built from source. After the
+tests finishes, it will delete the cluster._
+
+If you have already created your own Kubernetes cluster but haven't installed
+Knative, you can run with `test/e2e-tests.sh --run-tests`.
+
+If you have set up a running environment that meets
+[the e2e test environment requirements](#environment-requirements), you can run
+with `test/e2e-tests.sh --run-tests --skip-knative-setup`.
+
+## Running tests with `go test` command (recommended)
```

Comment:
I would drop "recommended". There's nothing wrong with using the scripts.

Reply:
Done.